### PR TITLE
Aqib/create-trip 

### PR DESCRIPTION
**Pull request by Muh-Aqib-Shah**
**Submit** button now works allowing logged In users to submit there trips from the create trips form, allowing the trip to be stored in db. Users are notified about form submission through a `toast`. Only logged In users can submit the form otherwise a "Not Logged In" `toast` is shown, in case of a error a Internal server error is sent by the api wich is handled and a erronous `toast` is shown to user.

### CODE REVIEW

File: .env.example
Comment (Muh-Hasan) [inline .env.example:1]: It was an example used to help others set up a project; please re-add the file.

File: src/components/create-trip/hotel-details.tsx [hunk 1]
```diff
@@ -207,8 +208,71 @@ const HotelDetails: React.FC<Props> = ({ stepfn, control }) => {
             <path d="M5 12l6 -6" />
           </svg>
         </Button>
-        <Button type="submit" className="w-[70%]">
-          Submit
+        <Button type="submit" disabled={btnState} className="w-[70%]">
+          {btnState ? (
+            <svg
```
Comment (Muh-Hasan) [inline src/components/create-trip/hotel-details.tsx:213]: Use the loader icon from the icon library and add the 'animate-spin' class to it.

**Review by Muh-Hasan — CHANGES_REQUESTED** [review]
- Add `create-trip` to the middleware file to prevent users from being redirected if they are not logged in.

- On submission, the previous button remains active, allowing the user to toggle between the forms; it should be disabled.

- Attach a route between '/trips' and '/create-trip' to the 'new trip' button on the '/trips' page.

- Use this toast: https://ui.shadcn.com/docs/components/sonner

**Comment by Muh-Aqib-Shah** [comment]
implemented the changes as you requested!

**Review by Muh-Hasan — APPROVED** [review]
Looks good!

**Pull request merged**